File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res= {0,0};
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++)
        {
            int x = nums[i];
            if(map.containsKey(target-x)){
                res[1] = i;
                res[0] = map.get(target-x);
                return res;
            }
            map.put(x,i);
        }
        return res;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> mp;
        for(int x: nums)
        {
            if(mp.find(x) != mp.end()){
                return true;
            }
            mp[x] = 1;
        }
        return false;
    }
};